Nodejs/Product/Nodejs/Project/NodeModulesNode.cs

```diff
@@ -27,9 +27,35 @@
 using Microsoft.VisualStudioTools.Project;
 using MessageBox = System.Windows.MessageBox;
 using Timer = System.Threading.Timer;
+using System.Threading.Tasks;
 
 namespace Microsoft.NodejsTools.Project {
     internal class NodeModulesNode : AbstractNpmNode {
+        /// <summary>
+        /// `INpmGlobalPackageProvider` that caches `IGlobalPackages` objects for reuse.
+        /// </summary>
+        private class CacheGlobalPackageProivder : INpmGlobalPackageProvider {
+            private Dictionary<string, WeakReference<IGlobalPackages>> _packages = new Dictionary<string, WeakReference<IGlobalPackages>>();
```

Comment:
Not really sure i see a significant enough advantage of using weak references here, especially since a strong reference is already being held onto by the project node and I don't anticipate a) the global folder frequently changing between projects and b) projects being loaded and unloaded often. Am I missing something?